### PR TITLE
[nnx] remove flagslib

### PR DESCRIPTION
**Pull request by cgarciae**
# What does this PR do?

This PR removes the `flagslib` module is favor of a `Module.set_attributes` method that recursively sets the attributes of all Modules in the Module graph. Basically, flags are now just Module attributes plus a mechanism to recursively set them.

### CODE REVIEW

File: flax/experimental/nnx/tests/test_module.py [hunk 1]
```diff
@@ -492,31 +516,31 @@ def __init__(self):
 
 class TestModuleDataclass:
   def test_basic(self):
-    @dataclasses.dataclass
+    @nnx.dataclass
```
Comment (chiamp) [inline flax/experimental/nnx/tests/test_module.py:519]: This test was modified by #3742.

File: flax/experimental/nnx/tests/test_pytree.py [hunk 2]
```diff
@@ -48,12 +47,12 @@ def __init__(self, y) -> None:
       pytree.x = 4
 
   def test_immutable_pytree_dataclass(self):
-    @dataclasses.dataclass(frozen=True)
+    @nnx.dataclass(frozen=True)
```
Comment (chiamp) [inline flax/experimental/nnx/tests/test_pytree.py:50]: This file was modified by #3742.

File: flax/linen/fp8_ops.py [hunk 3]
```diff
@@ -12,81 +12,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import dataclasses
-import numpy as np
 import warnings
```
Comment (chiamp) [inline flax/linen/fp8_ops.py:15]: These changes don't look related to this PR. Maybe rebasing will fix this?

File: flax/linen/linear.py [hunk 4]
```diff
@@ -31,12 +31,9 @@
 from jax import eval_shape, lax
 from jax.core import ShapedArray
 
-import opt_einsum
-
 from flax.core import meta
```
Comment (chiamp) [inline flax/linen/linear.py:34]: same here, rebase

File: flax/experimental/nnx/nnx/module.py [hunk 5]
```diff
@@ -452,6 +458,52 @@ def modules(self) -> tp.Iterator[tuple[Path, Module]]:
       if isinstance(value, Module):
         yield path, value
 
+  def set_attributes(
+    self, *filters: filterlib.Filter, **attributes: tp.Any
+  ) -> None:
+    """Sets the attributes of nested Modules including the current Module.
+    If the attribute is not found in the Module, it is ignored.
```
Comment (chiamp) [inline flax/experimental/nnx/nnx/module.py:465]: would you consider a flag that would raise an error if the attribute isn't found?

**Comment by cgarciae** [comment]
Cleaned the PR to remove the spurious changes.

File: flax/experimental/nnx/nnx/module.py (hunk 5 above)
Comment (cgarciae) [reply flax/experimental/nnx/nnx/module.py:462]: done
